packages/gatsby/src/utils/validate-page-component.ts

```diff
@@ -0,0 +1,105 @@
+import path from "path"
+import fs from "fs-extra"
+import { IGatsbyPage } from "../redux/types"
+
+const validationCache = new Map<string, boolean>()
+
+interface IErrorMeta {
+  id: string
+  context: Record<string, unknown>
+}
+
+export function validatePageComponent(
+  page: IGatsbyPage,
+  directory: string,
+  pluginName: string
+): { message?: string; error?: IErrorMeta; panicOnBuild?: boolean } {
+  const { component } = page
+  if (!component) {
+    throw new Error(`11322`)
+  }
+  if (validationCache.has(component)) {
+    return {}
+  }
+  if (!path.isAbsolute(component)) {
+    return {
+      error: {
+        id: `11326`,
+        context: {
+          pluginName: name,
```

Comment:
We are not passing `name` here from `createPage` action

Reply:
Weird that TypeScript didn't catch this. It seems to have randomly thought it was an ambient type from the dom lib